viadot/sources/vid_club.py

```diff
@@ -80,70 +81,134 @@ def build_query(
             )
         return url
 
+    def intervals(
+        self, from_date: str, to_date: str, days_interval: int
+    ) -> Tuple[List[str], List[str]]:
+        """
+        Breaks dates range into smaller by provided days interval.
+
+        Args:
+            from_date (str): Start date for the query.
+            to_date (str): End date for the query. By default, datetime.today() will be used.
+            days_interval (int): Days specified in date range per api call (test showed that 30-40 is optimal for performance).
+
+        Returns:
+            List[str], List[str]: Starts abd Ends lists that contains information about date ranges for specific period and time interval.
+        """
+
+        if to_date == None:
+            end_date = datetime.today().date()
+        else:
+            end_date = datetime.strptime(to_date, "%Y-%m-%d").date()
+        start_date = datetime.strptime(from_date, "%Y-%m-%d").date()
+        interval = timedelta(days=days_interval)
+        starts = []
+        ends = []
+        period_start = start_date
+        while period_start < end_date:
+            period_end = min(period_start + interval, end_date)
+            starts.append(period_start.strftime("%Y-%m-%d"))
+            ends.append(period_end.strftime("%Y-%m-%d"))
+            period_start = period_end
+
```

Comment:
"In case when somebody will put the same date as from_date and end_date or start_date younger than end_date, there will be empty Lists inside tuple" -> this situation is included in total_load function, but I also added logger for this situation in case someon will use this function separately.